### PR TITLE
Feat/automatic updates UI

### PR DESCRIPTION
**Pull request by matthme**
This PR improves the UI around automatic updates (issue https://github.com/holochain/launcher/issues/202) following Figma screens that were created a few weeks ago (not exactly 1:1 but quite similarly).

* it creates an "Updates" section in settings that only shows up if there is an update available
* it displays an orange dot on the settings gear if a new update is available
* it displays information about the update in the updates section like changelog, version number, release time and lets you download & install the update and restart launcher (with a loading indicator)

To test this PR:
* change the version field in the root level package.json to `0.400.0-rc.1`
* build the launcher executable locally with `yarn build:build:mac-arm64` or `yarn build:linux`etc.
* start launcher and verify that it offers to update to the latest compatible release (at the time of writing `0.400.0-rc.2`)

### CODE REVIEW

File: src/renderer/src/routes/settings/(menu)/[slug]/components/LauncherUpdates.svelte [hunk 1]
```diff
@@ -0,0 +1,96 @@
+<script lang="ts">
+	import TimeAgo from 'javascript-time-ago';
+
+	import { Button } from '$components';
+	import { markdownParseSafeTailwind } from '$helpers';
+	import { i18n, trpc } from '$services';
+	import { ProgressBar, getToastStore } from '@skeletonlabs/skeleton';
+
+	// import { getModalStore } from '@skeletonlabs/skeleton';
```
Comment (c12i) [inline src/renderer/src/routes/settings/(menu)/[slug]/components/LauncherUpdates.svelte]: Is this code still needed?

File: src/renderer/src/routes/settings/(menu)/[slug]/components/LauncherUpdates.svelte [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+	const client = trpc();
+	const toastStore = getToastStore();
+
+	// 	export interface ProgressInfo {
+	//     total: number;
+	//     delta: number;
+	//     transferred: number;
+	//     percent: number;
+	//     bytesPerSecond: number;
+	// }
```
Comment (c12i) [inline src/renderer/src/routes/settings/(menu)/[slug]/components/LauncherUpdates.svelte]: Same here?

File: src/renderer/src/routes/settings/components/RegularMenu.svelte [hunk 3]
```diff
@@ -81,11 +92,38 @@
...
+		console.log(Date.now(), 'LAUNCHER UPDATE AVAILABLE CHANGED: ', val);
+		if (val.isSuccess && val.data) {
+			if (!menuEntries.map((entry) => entry.name).includes($i18n.t(LAUNCHER_UPDATES))) {
+				console.log('pushing menu entry.');
+				const newMenuEntries = menuEntries;
+				newMenuEntries.unshift(launcherUpdateMenuEntry);
+				menuEntries = newMenuEntries;
+			}
+		}
+	});
```
Comment (c12i) [inline src/renderer/src/routes/settings/components/RegularMenu.svelte]:
```suggestion
if (val.isSuccess && val.data && !menuEntries.map((entry) => entry.name).includes($i18n.t(LAUNCHER_UPDATES))) {
    menuEntries = [launcherUpdateMenuEntry, ...menuEntries];
}
```

File: src/renderer/src/lib/helpers/other.ts [hunk 4]
```diff
@@ -236,3 +238,21 @@ export const resizeImage = async (file: File): Promise<Uint8Array | null> => {
...
+	const markedData = marked.parse(input) as string;
+	const htmlString = DOMPurify.sanitize(markedData);
+	return htmlString
+		.replaceAll('<ul>', '<ul class="list-disc ml-6">')
+		.replaceAll('<ol>', '<ol class="list-decimal ml-6">')
+		.replaceAll('<h1>', '<h1 class="h1">')
+		.replaceAll('<h2>', '<h1 class="h2">')
+		.replaceAll('<h3>', '<h1 class="h3">')
+		.replaceAll('<h4>', '<h1 class="h4">')
+		.replaceAll('<h5>', '<h1 class="h5">')
```
Comment (c12i) [inline src/renderer/src/lib/helpers/other.ts]: Are these tags still supposed to be replaced with `h1`?

**Review by c12i — CHANGES_REQUESTED** [review]

**Comment by matthme** [comment]
Good points, thank you! I addressed them all.

**Review by c12i — APPROVED** [review]

**Pull request merged**